### PR TITLE
src: remove some unused parameters

### PR DESCRIPTION
**Pull request by catenacyber**
Link to ticket: https://redmine.openinfosecfoundation.org/issues/
None, just some cleanup

Describe changes:
- src: remove some unused function parameters

Found with `-Wunused-parameter` and focusing on static functions

Incomplete, but still useful

Inspired by https://github.com/OISF/suricata/pull/11026#discussion_r1600574165

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 20643

File: rust/src/http2/http2.rs [hunk 1]
```diff
@@ -275,6 +275,7 @@ impl HTTP2Transaction {
                     range::http2_range_append(sfcm, self.file_range, decompressed)
                 }
             }
+            println!("lolh {} {}", decompressed.len(), input.len());
```
Comment (jasonish) [inline rust/src/http2/http2.rs:278]: lol

File: src/app-layer-htp.c [hunk 2]
```diff
@@ -2549,10 +2549,8 @@ static void HTPConfigSetDefaultsPhase1(HTPCfgRec *cfg_prec)
      * config, we have to set the soft limit as well. If libhtp starts using
      * the soft limit in the future, we at least make sure we control what
      * it's value is. */
-    htp_config_set_field_limits(cfg_prec->cfg,
-            (size_t)HTP_CONFIG_DEFAULT_FIELD_LIMIT_SOFT,
+    htp_config_set_field_limits(cfg_prec->cfg, (size_t)HTP_CONFIG_DEFAULT_FIELD_LIMIT_SOFT,
             (size_t)HTP_CONFIG_DEFAULT_FIELD_LIMIT_HARD);
```
Comment (jasonish) [inline src/app-layer-htp.c:2553]: Add some params as well?

Comment (catenacyber) [reply src/app-layer-htp.c:2553]: No param is added.
This is just clang-format doing

**Comment by catenacyber** [comment]
Continued on https://github.com/OISF/suricata/pull/11097

**Pull request closed without merging**